test/net_http_test.rb

```diff
@@ -0,0 +1,241 @@
+require 'test_helper'
+require 'semian/net_http'
+require 'thin'
+
+class TestNetHTTP < MiniTest::Unit::TestCase
+  class RackServer
+    def self.call(env)
+      response_code = env['REQUEST_URI'].delete("/")
+      response_code = '200' if response_code == ""
+      [response_code, {'Content-Type' => 'text/html'}, ['Success']]
+    end
+  end
+
+  def setup
+    # create a simple rack server so we can use toxiproxy
+    start_server_singleton
+    Semian["http_localhost_41234"].reset if Semian["http_localhost_41234"]
+    Semian["http_localhost_41235"].reset if Semian["http_localhost_41235"]
```

Comment:
These ports are in the Linux ephemeral port range:

```
simon@sb1.chi2:~ 1$ cat /proc/sys/net/ipv4/ip_local_port_range 
32768	61000
```

This means you can experience intermittent failures because an ephemeral port may already be started on this port. Just make it `< 32768` and you're good.